cmd/contour/serve.go

```diff
@@ -275,6 +278,28 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 		FieldLogger: log.WithField("context", "endpointstranslator"),
 	}
 
+	// snapshotCache is used to store the state of what all xDS services should
+	// contain at any given point in time.
+	snapshotCache := cache.NewSnapshotCache(false, cache.IDHash{},
+		log.WithField("context", "XDSServer"))
```

Comment:
I think it's better to use a specialized node ID hasher to avoid taking a dependency on the Envoy configuration. For example:
```Go

package xds

type ConstantHash string

func (c ConstantHash) ID(*envoy_config_core_v3.Node) string {
         return string(c)
}

func (c ConstantHash) String() string {
         return string(c)
}

var _ cache.NodeHash = ConstantHash("")
var DefaultHash = ConstantHash("contour") 

```

Usage looks like:
```Go
s := cache.NewSnapshotCache(false, xds.DefaultHash, log)
...
s.SetSnapshot(xds.DefaultHash.String(), snapshot)
```

Reply:
Neat I see how you'd use this now. I wasn't ever setting the hash in the `cache.NewSnapshotCache(...)`. I've got this wired in and looks like it's working!